### PR TITLE
feat(issue-platform): Add occurrence data to kafka messaging

### LINKED ISSUE #42529: Update post process forwarder to  work with generic issues
We're currently sending generic events to snuba via eventstream. We also want to be able to schedule `post_process_group` via the [post process forwarder](https://github.com/getsentry/sentry/blob/5d006832f3fa640a6927bcbf38b41bbbcf3388fa/src/sentry/eventstream/kafka/postprocessworker.py#L94-L122).

This might just work as is - we will already grab the `group_id` and `is_new`, etc parameters from eventstream. If this does just work, then we need to just add an additional celery worker to process the `post_process_group_generic` queue.

Note that we don't want to use the list of `GroupState` objects here. That's specific to performance issues, and we'll probably phase it out after we move them over to the issue platform.

### PR DESCRIPTION
**Pull request by rjo100**
Adds occurrence id + data to the Kafka processing

Closes https://github.com/getsentry/sentry/issues/42529

### CODE REVIEW

File: src/sentry/issues/ingest.py [hunk 1]
```diff
@@ -210,6 +210,7 @@ def send_issue_occurrence_to_eventstream(
         primary_hash=occurrence.fingerprint[0],
         received_timestamp=group_event.data.get("received") or group_event.datetime,
         skip_consume=False,
+        group_id=group_info.group.id,
```
Comment (wedamija) [inline src/sentry/issues/ingest.py:213]: Why do we need to pass this now?

**Review by wedamija — COMMENTED** [review]
Looks good, could you update some tests to validate this?

Probably these 3 tests: https://github.com/getsentry/sentry/blob/80f51d94c6abd56c77f8f0fb42bf75e273fbd61b/tests/sentry/eventstream/test_eventstream.py#L273
and https://github.com/getsentry/sentry/blob/80f51d94c6abd56c77f8f0fb42bf75e273fbd61b/tests/sentry/eventstream/test_eventstream.py#L218

File: src/sentry/issues/ingest.py (hunk 1 above)
Comment (rjo100) [reply src/sentry/issues/ingest.py:213]: I guess I just misread the GH issue details. Will remove

**Comment by rjo100** [comment]
Not sure what exactly is needed in those tests? Just to see that we're not passing `occurrence_id` in the headers?

**Comment by wedamija** [comment]
> Not sure what exactly is needed in those tests? Just to see that we're not passing `occurrence_id` in the headers?

Just validate that occurrence_id is correct (or not present) depending on the test

**Review by wedamija — APPROVED** [review]

**Pull request merged**